tests/Doctrine/Tests/Models/Issue5887/Issue5887CustomIdObjectType.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace Doctrine\Tests\Models\Issue5887;
+
+use Doctrine\DBAL\Platforms\AbstractPlatform;
+use Doctrine\DBAL\Types\Type;
+
+class Issue5887CustomIdObjectType extends Type
```

Comment:
It can also inherit `StringType` instead so you don't need to implement the method `getSQLDeclaration()`.